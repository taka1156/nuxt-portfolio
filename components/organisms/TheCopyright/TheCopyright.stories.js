import TheCopyright from './TheCopyright.vue';
import { dummyCopyrightUrl } from '__testdata__/testdata.js';

export default {
  title: 'Organisms/TheCopyright',
  component: TheCopyright
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { TheCopyright },
  template: '<the-copyright v-bind="$props" />'
});

export const Default = Template.bind({});
Default.args = { copyrightUrl: dummyCopyrightUrl };
