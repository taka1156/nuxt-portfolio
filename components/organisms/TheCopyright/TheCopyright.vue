<template>
  <div>
    <div class="the-copyright">
      <base-text>
        &copy; 2019 taka1156 <br />使用素材、プライバシーポリシー等は
        <base-link :route-to="copyrightUrl" :is-target="true">コチラ</base-link>
      </base-text>
    </div>
  </div>
</template>

<script>
import BaseLink from '../../atoms/BaseLink/BaseLink';
import BaseText from '../../atoms/BaseText/BaseText';

export default {
  name: 'TheCopyright',
  components: {
    'base-link': BaseLink,
    'base-text': BaseText
  },
  props: {
    /**
     * ポリシーサイトのURL
     */
    copyrightUrl: {
      type: String,
      required: true
    }
  }
};
</script>

<style scoped>
::v-deep .base-text--extend,
a {
  color: white;
}

.the-copyright {
  width: 100%;
  padding: 0;
  margin: 0;
  text-align: left;
  background-color: rgba(40, 167, 69, 0.9);
}
</style>
