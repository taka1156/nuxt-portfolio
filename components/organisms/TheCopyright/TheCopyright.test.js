import { mount } from '@vue/test-utils';
import TheCopyright from './TheCopyright.vue';

describe('TheNavigation', () => {
  const theCopyright = mount(TheCopyright);

  it('スナップショット', () => {
    expect(theCopyright.html()).toMatchSnapshot();
  });
});
