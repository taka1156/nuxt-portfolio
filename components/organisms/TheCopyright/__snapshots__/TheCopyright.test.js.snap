// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheNavigation 値がDOMに反映されているか 1`] = `
"<div>
  <div class=\\"the-copyright\\">
    <p class=\\"base-text--extend\\">
      © 2019 taka1156 <br>使用素材、プライバシーポリシー等は
      <a href=\\"http://placehold.jp\\" tag=\\"a\\" target=\\"_blank\\" rel=\\"noopener noreferrer\\" class=\\"base-link--extend\\">コチラ</a></p>
  </div>
</div>"
`;
